### PR TITLE
Support for Counter

### PR DESCRIPTION
**Pull request by tatry**
* Unify some names in API
* Support for counter read/set/reset
* Support for indexing counter by structure
* Minor code reformat

### CODE REVIEW

File: CLI/counter.c [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+    psabpf_counter_ctx_init(&ctx);
+    psabpf_counter_entry_init(&entry);
+
+    if (parse_pipeline_id(&argc, &argv, &psabpf_ctx) != NO_ERROR)
+        goto clean_up_psabpf;
+
+    if (parse_dst_counter(&argc, &argv, &counter_name, &psabpf_ctx, &ctx) != NO_ERROR)
+        goto clean_up_psabpf;
+
+    bool counter_key_provided = false;
```
Comment (kmateuszssak) [inline CLI/counter.c:231]: Could be also 
```
bool counter_key_provided = (argc >= 1 && is_keyword(*argv, "key"));
if (counter_key_provided) {...
```

File: CLI/counter.c [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+        if (parse_counter_key(&argc, &argv, &entry) != NO_ERROR)
+            goto clean_up_psabpf;
+    }
+
+    if (argc > 0) {
+        fprintf(stderr, "%s: unused argument\n", *argv);
+        goto clean_up_psabpf;
+    }
+
+    json_t *root = json_object();
```
Comment (kmateuszssak) [inline CLI/counter.c:243]: Could we separate this json stuff from `do_counter_get`? Maybe move this part to method like `get_json_counter_data` ?

File: CLI/counter.c [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+
+    if (parse_counter_value(&argc, &argv, &ctx, &entry) != NO_ERROR)
+        goto clean_up;
+
+    if (argc > 0) {
+        fprintf(stderr, "%s: unused argument\n", *argv);
+        goto clean_up;
+    }
+
+    ret = psabpf_counter_set(&ctx, &entry);
```
Comment (kmateuszssak) [inline CLI/counter.c:339]: This is beautiful👍

**Review by kmateuszssak — COMMENTED** [review]
Just two questions.

File: CLI/counter.c (hunk 1 above)
Comment (tatry) [reply CLI/counter.c]: Done

File: CLI/counter.c (hunk 2 above)
Comment (tatry) [reply CLI/counter.c:217]: Done

**Pull request merged**